library/Core/src/Juvix/Core/IR/Types/Base.hs

```diff
@@ -244,8 +362,10 @@ deriving instance
   NFData (AbstractWith ty primTy primVal)
 
 data GlobalWith ty ext primTy primVal
-  = GDatatype (DatatypeWith ty primTy primVal)
-  | GDataCon (DataConWith ty primTy primVal)
+  = GDatatype (Datatype' ext primTy primVal)
+  | RawGDatatype (RawDatatype' ext primTy primVal)
+  | GDataCon (DataCon' ext primTy primVal)
+  | RawGDataCon (RawDataCon' ext primTy primVal)
   | GFunction (FunctionWith ty ext primTy primVal)
   | GAbstract (AbstractWith ty primTy primVal)
```

Comment:
Would this work better as separate `Global` and `RawGlobal` (also containing `RawFunction` and `RawAbstract`, analogous to the data stuff) types and just removing the `ty` argument completely? It seems that at any given point you are only consuming/producing either all raw things, or all checked things. If this would work it would also fix the exhaustiveness warnings I think.

Reply:
I was wondering if you would want that. As I mentioned, I think removing `ty` completely would simplify things. I'll go ahead and remove it then.

Reply:
actually could I roll back to [this](https://github.com/heliaxdev/juvix/pull/668/commits/8ed00d70720f41df06b6e51d3bd3b9684a860f1c) commit and merge to develop so that all other changes are in? Especially with #697 coming. Removing `ty` the rest of the way can come in a later PR.

Reply:
Yeah that should be fine I think! Sorry I ended up having to touch the same code as you.